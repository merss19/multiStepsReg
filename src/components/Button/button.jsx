"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classNames = require("classnames");
var ButtonTypes;
(function (ButtonTypes) {
    ButtonTypes["info"] = "secondary";
    ButtonTypes["error"] = "error";
    ButtonTypes["success"] = "primary";
})(ButtonTypes = exports.ButtonTypes || (exports.ButtonTypes = {}));
exports.Button = function (_a) {
    var children = _a.children, onClick = _a.onClick, type = _a.type, styleBtn = _a.styleBtn, wide = _a.wide, prefix = _a.prefix;
    var classes = classNames(prefix, (_b = {},
        _b[prefix + "--" + styleBtn] = true,
        _b['wide'] = wide,
        _b));
    return (<button type={type} className={classes} onClick={onClick}>
      {children}
    </button>);
    var _b;
};
exports.Button.defaultProps = {
    styleBtn: ButtonTypes.info,
    type: 'button',
    wide: false,
    prefix: 'btn'
};
