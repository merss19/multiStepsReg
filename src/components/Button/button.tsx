import * as React from 'react';
import * as classNames from 'classnames';

export type ButtonType = ButtonTypes.info | ButtonTypes.error | ButtonTypes.success;

export enum ButtonTypes {
  info = 'secondary',
  error = 'error',
  success = 'primary'
}

interface OwnProps {
  onClick?: (data?: any) => any
  styleBtn?: ButtonType,
  type?: string,
  wide?: boolean
  prefix?: string
}

export const Button: React.SFC<OwnProps> = ({children, onClick, type, styleBtn, wide, prefix}) => {
  const classes = classNames(prefix, {
    [`${prefix}--${styleBtn}`]: true,
    ['wide']: wide
  });
  return (
    <button
      type={type}
      className={classes}
      onClick={onClick}
    >
      {children}
    </button>)
};

Button.defaultProps = {
  styleBtn: ButtonTypes.info,
  type: 'button',
  wide: false,
  prefix: 'btn'
};