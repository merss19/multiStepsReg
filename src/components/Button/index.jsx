"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var button_1 = require("./button");
exports.Button = button_1.Button;
exports.ButtonTypes = button_1.ButtonTypes;
