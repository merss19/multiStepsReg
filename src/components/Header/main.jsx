"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("components/Button");
var react_router_dom_1 = require("react-router-dom");
var storage_1 = require("tools/storage");
var Header = /** @class */ (function (_super) {
    __extends(Header, _super);
    function Header() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Header.prototype.logOut = function () {
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        storage_1.storage.removeState('auth');
    };
    Header.prototype.render = function () {
        return (<div className='header header--hero'>
        <div className='grid header__inner'>
          <div className='grid__cell header__burger'>
            <span className='header__ico-burger'>
              <svg className='svg-icon ico-burger'>
                <use xlinkHref="#ico-burger"></use>
              </svg>
            </span>
          </div>

          <h1 className='grid__cell header__logo'>
            Ясегодня
            <svg className='svg-icon ys-logo-web'>
              <use xlinkHref="#ys-logo-web"></use>
            </svg>
          </h1>

          <div className='grid__cell header__right-side'>
            <Button_1.Button type='button' styleBtn={Button_1.ButtonTypes.info}>
              <react_router_dom_1.Link to='/' onClick={this.logOut.bind(this)}>Выход</react_router_dom_1.Link>
            </Button_1.Button>
          </div>
        </div>
      </div>);
    };
    return Header;
}(React.Component));
exports.Header = Header;
exports.default = Header;
