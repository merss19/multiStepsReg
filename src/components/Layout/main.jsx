"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Background_1 = require("../Background");
var LogoLink_1 = require("../LogoLink");
var SVGInjector = require("svg-injector");
var Header_1 = require("components/Header");
var Layout = /** @class */ (function (_super) {
    __extends(Layout, _super);
    function Layout() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Layout.prototype.componentDidMount = function () {
        this.svgInject();
    };
    Layout.prototype.svgInject = function () {
        var sprite = {
            body: document.querySelector('body'),
            img: document.createElement('img')
        };
        sprite.img.className = 'injected-svg';
        sprite.img.id = 'svg-inject-me';
        sprite.img.src = process.env.PUBLIC_URL + '/img/symbol-sprite.svg';
        sprite.body.appendChild(sprite.img);
        var mySVGsToInject = document.querySelector('#svg-inject-me');
        SVGInjector(mySVGsToInject);
    };
    Layout.prototype.renderLogo = function () {
        if (!this.props.notLogo) {
            return (<div className='grid entry-header'>
          <div className='grid__cell todayme-logo'>
            <LogoLink_1.default />
          </div>
        </div>);
        }
    };
    Layout.prototype.render = function () {
        var _a = this.props, children = _a.children, type = _a.type;
        if (type === 'main') {
            return (<Header_1.Header />);
        }
        else {
            return (<div className='layout layout--entry'>
          {this.renderLogo()}
          {children}
          <Background_1.default />
        </div>);
        }
    };
    Layout.defaultProps = {
        notLogo: false,
        type: 'signUP'
    };
    return Layout;
}(React.Component));
exports.Layout = Layout;
exports.default = Layout;
