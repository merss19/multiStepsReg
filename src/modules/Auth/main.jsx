"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_1 = require("react-router");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var CustomModal_1 = require("modules/CustomModal");
var SignUp_1 = require("modules/SignUp");
var storage_1 = require("tools/storage");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state, ownProps) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isAuth: selectors_1.selectIsAuth(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    user: selectors_1.selectUser(state)
}); };
var ProtectedRouteComponent = /** @class */ (function (_super) {
    __extends(ProtectedRouteComponent, _super);
    function ProtectedRouteComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProtectedRouteComponent.prototype.componentWillMount = function () {
        var _this = this;
        var checkAuth = this.props.checkAuth;
        checkAuth(function (data) { return _this.checkAuth(data); }, function () { return _this.leave(); });
    };
    ProtectedRouteComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching ||
            nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    ProtectedRouteComponent.prototype.leave = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('txId', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('packageType', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.checkAuth = function (data) {
        var _a = this.props, changeStep = _a.changeStep, history = _a.history;
        if (data.paidState !== 0) {
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
                history.replace('/signup');
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else if (!data.emailConfirmed) {
                return null;
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            changeStep(SignUp_1.Steps.two);
        }
        else {
            changeStep(SignUp_1.Steps.four);
        }
    };
    ProtectedRouteComponent.prototype.modalClick = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.clickHandler = function () {
        var isError = this.props.isError;
        if (!isError) {
            return null;
        }
        return this.modalClick();
    };
    ProtectedRouteComponent.prototype.renderModal = function (modal) {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg, isError = _a.isError;
        return (<CustomModal_1.CustomModal modal={modal} resultText={errMsg} clickHandler={this.clickHandler.bind(this)} isLoader={isFetching}/>);
    };
    ProtectedRouteComponent.prototype.renderContent = function (props) {
        var _a = this.props, ProtectedComponent = _a.component, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<div>
        <ProtectedComponent {...props}/>
        {this.renderModal(CustomModal_1.ModalTypes.error)}
      </div>);
    };
    ProtectedRouteComponent.prototype.renderRoute = function (props) {
        var isAuth = this.props.isAuth;
        return (<div>
        {isAuth ?
            this.renderContent(props) :
            this.renderModal(CustomModal_1.ModalTypes.info)}
      </div>);
    };
    ProtectedRouteComponent.prototype.render = function () {
        var _this = this;
        var _a = this.props, ProtectedComponent = _a.component, isAuth = _a.isAuth, isError = _a.isError, isFetching = _a.isFetching, errMsg = _a.errMsg, rest = __rest(_a, ["component", "isAuth", "isError", "isFetching", "errMsg"]);
        return (<react_router_dom_1.Route {...rest} render={function (props) {
            return _this.renderRoute(props);
        }}/>);
    };
    return ProtectedRouteComponent;
}(React.Component));
exports.ProtectedRouteComponent = ProtectedRouteComponent;
exports.ProtectedRoute = react_redux_1.connect(mapStateToProps, {
    checkAuth: ducks_1.checkAuth,
    changeStep: SignUp_1.changeStep,
    toggleModal: CustomModal_1.toggleModal
})(ProtectedRouteComponent);
exports.default = react_router_1.withRouter(exports.ProtectedRoute);
