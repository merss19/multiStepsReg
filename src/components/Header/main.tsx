import * as React from 'react';
import {Button, ButtonTypes} from "components/Button";
import {Link} from "react-router-dom";
import {storage} from "tools/storage";

interface OwnProps {
}

export class Header extends React.Component<OwnProps, {}> {

  logOut() {
    storage.remove('token', {path: '/'});
    storage.remove('program', {path: '/'});
    storage.remove('promoName', {path: '/'});
    storage.removeState('auth');
  }

  render() {
    return (
      <div className='header header--hero'>
        <div className='grid header__inner'>
          <div className='grid__cell header__burger'>
            <span className='header__ico-burger'>
              <svg className='svg-icon ico-burger'>
                <use xlinkHref="#ico-burger"></use>
              </svg>
            </span>
          </div>

          <h1 className='grid__cell header__logo'>
            Ясегодня
            <svg className='svg-icon ys-logo-web'>
              <use xlinkHref="#ys-logo-web"></use>
            </svg>
          </h1>

          <div className='grid__cell header__right-side'>
            <Button
              type='button'
              styleBtn={ButtonTypes.info}
            >
              <Link to='/' onClick={this.logOut.bind(this)}>Выход</Link>
            </Button>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;