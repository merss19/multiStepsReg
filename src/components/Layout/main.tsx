import * as React from 'react';
import Background from '../Background';
import LogoLink from '../LogoLink'
import * as SVGInjector from 'svg-injector'
import {Header} from 'components/Header';

interface Sprite {
  readonly body: any
  readonly img: HTMLImageElement
}
interface OwnProps {
  notLogo?: boolean
  type: 'main' | 'signUP'
}
export class Layout extends React.Component<OwnProps, {}> {

  static defaultProps = {
    notLogo: false,
    type:'signUP'
  };

  componentDidMount() {
    this.svgInject()
  }

  svgInject(): void {
    const sprite: Sprite = {
      body: document.querySelector('body'),
      img: document.createElement('img')
    };
    sprite.img.className = 'injected-svg';
    sprite.img.id = 'svg-inject-me';
    sprite.img.src = process.env.PUBLIC_URL + '/img/symbol-sprite.svg';
    sprite.body.appendChild(sprite.img);
    let mySVGsToInject: any = document.querySelector('#svg-inject-me');
    SVGInjector(mySVGsToInject);
  }

  renderLogo() {
    if (!this.props.notLogo) {
      return (
        <div className='grid entry-header'>
          <div className='grid__cell todayme-logo'>
            <LogoLink/>
          </div>
        </div>
      )
    }
  }

  render() {
    const {children, type} = this.props;
    if(type === 'main'){
      return (<Header/>);
    } else {
      return (
        <div className='layout layout--entry'>
          {this.renderLogo()}
          {children}
          <Background/>
        </div>
      );
    }
  }
}

export default Layout;