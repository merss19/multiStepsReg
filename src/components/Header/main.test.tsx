import {shallow} from 'enzyme';
import {Header} from './main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {Button} from "../Button/button";
import * as sinon from 'sinon';
import {storage} from "../../tools/storage/index";
import {Link} from "react-router-dom";

configure({adapter: new Adapter()});

const start = () => {

  const buildComponent = () => {
    return shallow(<Header />)
  };
  return {
    buildComponent
  }
};
describe('Component Header', () => {
  it('renders component', () => {
    const component = start().buildComponent();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(start().buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('render Button', () => {
    const component = start().buildComponent();
    expect(component.find(Button)).toHaveLength(1);
  });
  it('click Button', () => {
    const spy = sinon.spy(Header.prototype, 'logOut');
    const component = start().buildComponent();
    const button = component.find(Link);
    button.simulate('click');
    expect(spy.called).toBe(true);
  });
  it('call logOut', () => {
    storage.save('token', 'test');
    storage.save('promoName', 'tele2');
    storage.save('program', 'hero');
    storage.saveState('auth', 'sdfsdf');
    const component = start().buildComponent();
    const instance = component.instance() as Header;
    instance.logOut();
    expect(storage.load('token')).toBeUndefined();
    expect(storage.load('program')).toBeUndefined();
    expect(storage.load('promoName')).toBeUndefined();
    expect(storage.loadState('auth')).toBeUndefined();
  });
});
