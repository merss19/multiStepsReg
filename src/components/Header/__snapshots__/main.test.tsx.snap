// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header renders correctly 1`] = `
<div
  className="header header--hero"
>
  <div
    className="grid header__inner"
  >
    <div
      className="grid__cell header__burger"
    >
      <span
        className="header__ico-burger"
      >
        <svg
          className="svg-icon ico-burger"
        >
          <use
            xlinkHref="#ico-burger"
          />
        </svg>
      </span>
    </div>
    <h1
      className="grid__cell header__logo"
    >
      Ясегодня
      <svg
        className="svg-icon ys-logo-web"
      >
        <use
          xlinkHref="#ys-logo-web"
        />
      </svg>
    </h1>
    <div
      className="grid__cell header__right-side"
    >
      <Component
        prefix="btn"
        styleBtn="secondary"
        type="button"
        wide={false}
      >
        <Link
          onClick={[Function]}
          replace={false}
          to="/"
        >
          Выход
        </Link>
      </Component>
    </div>
  </div>
</div>
`;
